onnxruntime/test/providers/cpu/nn/batch_norm_op_test.cc

```diff
@@ -23,19 +23,27 @@ void TestBatchNorm(const InputDataMap& input_data_map,
                    const vector<int64_t>& expected_output_shape,
                    int64_t spatial_mode = 1,
                    OpTester::ExpectResult expect_result = OpTester::ExpectResult::kExpectSuccess,
-                   const std::string& err_str = "") {
-  OpTester test("BatchNormalization");
+                   const std::string& err_str = "",
+                   int opset_version = 9) {
+  OpTester test("BatchNormalization", opset_version);
   if (epsilon.has_value()) {
     test.AddAttribute("epsilon", epsilon.value());
   }
-  test.AddAttribute("spatial", spatial_mode);
+  if (opset_version < 9) {  // spatial is only defined for opset-8 and below in the spec
+    test.AddAttribute("spatial", spatial_mode);
+  }
   test.AddInput<float>("X", input_shapes_map.at("X"), input_data_map.at("X"));
   test.AddInput<float>("scale", input_shapes_map.at("scale"), input_data_map.at("scale"));
   test.AddInput<float>("B", input_shapes_map.at("B"), input_data_map.at("B"));
   test.AddInput<float>("mean", input_shapes_map.at("mean"), input_data_map.at("mean"));
   test.AddInput<float>("var", input_shapes_map.at("var"), input_data_map.at("var"));
   test.AddOutput<float>("output", expected_output_shape, expected_output);
-  test.Run(expect_result, err_str, {kTensorrtExecutionProvider});// Weight as input is not supported by TensorRT
+  // Weight as input is not supported by TensorRT and spatial == 0 is not supported by Nuphar
+  std::unordered_set<std::string> excluded_eps = {kCudaExecutionProvider, kTensorrtExecutionProvider};
```

Comment:
>kCudaExecutionProvider [](start = 50, length = 22)

spatial == 0 should be supported by CUDA

Reply:
Thanks for catching. Added that incorrectly.

Reply:
Fixed